### PR TITLE
feat: allow setting native hls and dash playback

### PR DESCRIPTION
**Pull request by OrenMe**
### Description of the Changes

By default we use HLS and Dash adapters for playback of HLS and Dash streams.
This feature will allow setting the required adapter to be native or not, so we can use native HLS playback on Safari for example.

```json
{
  "playback": {
    "preferNative": {
      "hls": false,
      "dash": false
    }
  }
}
```

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [x] new unit / functional tests have been added (whenever applicable)
- [x] test are passing in local environment 
- [x] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/player-config.json [hunk 1]
```diff
@@ -7,14 +7,16 @@
     "preload": "none",
     "autoplay": false,
     "muted": false,
-    "options": {
+    "tweaks": {
```
Comment (dan-ziv) [inline src/player-config.json]: @OrenMe  if we're changing this key we should change also the reference in dash & hls adapters

File: src/engines/html5/media-source/adapters/native-adapter.js [hunk 2]
```diff
@@ -70,11 +77,15 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
    * Checks if NativeAdapter can play a given mime type.
    * @function canPlayType
    * @param {string} mimeType - The mime type to check
+   * @param {boolean} preferNative - prefer native flag
```
Comment (yairans) [inline src/engines/html5/media-source/adapters/native-adapter.js]: you can annotate the default value true
`@param {boolean} [preferNative=true] - prefer native flag
`

File: src/engines/html5/media-source/media-source-provider.js [hunk 3]
```diff
@@ -65,15 +65,16 @@ export default class MediaSourceProvider {
   /**
    * Checks if the a media source adapter can play a given source.
    * @param {Source} source - The source object to check.
+   *  @param {boolean} preferNative - prefer native flag
```
Comment (yairans) [inline src/engines/html5/media-source/media-source-provider.js]: as above

File: src/engines/html5/media-source/media-source-provider.js [hunk 4]
```diff
@@ -95,7 +96,7 @@ export default class MediaSourceProvider {
   static getMediaSourceAdapter(videoElement: HTMLVideoElement, source: Source, config: Object): ?IMediaSourceAdapter {
     if (videoElement && source && config) {
       if (!MediaSourceProvider._selectedAdapter) {
-        MediaSourceProvider.canPlaySource(source);
+        MediaSourceProvider.canPlaySource(source, false);
```
Comment (yairans) [inline src/engines/html5/media-source/media-source-provider.js]: why false?

File: src/player-config.json [hunk 5]
```diff
@@ -8,14 +8,16 @@
...
```
Comment (yairans) [inline src/player-config.json]: i think 'options' is more intuitive...

File: src/engines/html5/media-source/adapters/native-adapter.js (hunk 2 above)
Comment (OrenMe) [reply src/engines/html5/media-source/adapters/native-adapter.js]: Nice, didn't know this 👍

File: src/engines/html5/media-source/media-source-provider.js (hunk 4 above)
Comment (OrenMe) [reply src/engines/html5/media-source/media-source-provider.js]: why do we even have this check here? we should be getting here after we already have a selectedAdapter anyway.
I'll change to true for consistency, but this seems unnecessary.

Comment (yairans) [reply src/engines/html5/media-source/media-source-provider.js]: This is api. You can asssume nothing how getting here. 
Why true/false. You have the param from config

**Pull request merged**